### ISSUE TITLE
Adding a Fedora-based distro to CI?

### ISSUE THREAD

**Issue by workingjubilee** [tcdi/pgx#782]
It's becoming pretty obvious that Fedora has a lot of eccentricities compared to Debian (or Debian has a lot of eccentricities compared to Fedora, take your pick...). We might be best off having an instance of a distro derived from Fedora (which is the ultimate upstream of RHEL, CentOS Stream, and a lot of others...) in our CI, even if it's just on the nightly schedule instead of gating every PR.

**Comment by BradyBonnette** [comment]
💯

**Comment by workingjubilee** [comment]
We discussed this during our weekly.
Currently Red Hat has:
Fedora Linux as the upstream of
CentOS Stream as the upstream of
Red Hat Enterprise Linux.
Note that CentOS and CentOS Stream are _not_ the same thing.

As Red Hat controls all those pieces, we can assume that if we break one, we break them all, so we want one of them in our CI. And Amazon Linux seems to be heavily RHEL-based, too.

**Comment by BradyBonnette** [comment]
As of a few minutes ago, I was able to prove temporarily reverting https://github.com/tcdi/pgx/commit/d5fb78fa3674e81654bdf5a5c584f214d9b71916 locally and running the test suite it in a Fedora:36 container fails the build as we would expect. Reverting the revert (i.e. changing it back to current `develop`) and re-running the suite in the container succeeds the build as we would expect.

So Ill set up some CI stuff to make this happen. At first I am going to see just how long it takes to build via containers in Github Actions, and if it's not significantly longer then we may spin these up in a normal pull request flow considering the number of issues we got when the allowlist stuff broke Fedora-based builds. However, if Fedora container builds take a long time, we may just put these in the nightly "will-it-blend" setup.
